src/main/kotlin/no/nav/syfo/Environment.kt

```diff
@@ -3,133 +3,172 @@ package no.nav.syfo
+data class AppEnv(
     val applicationPort: Int,
     val applicationThreads: Int,
-    val syfosyketilfelleUrl: String,
-    val syfosmregisterUrl: String,
-    val syfosmregisterScope: String,
-    val narmestelederUrl: String,
-    val narmestelederScope: String,
-    val aadAccessTokenUrl: String,
+    val remote: Boolean = false,
+    val runningInGCPCluster: Boolean
+)
+
+data class AuthEnv(
+    val serviceuserUsername: String,
+    val serviceuserPassword: String,
     val clientId: String,
     val clientSecret: String,
+    val aadAccessTokenUrl: String,
     val loginserviceDiscoveryUrl: String,
-    val loginserviceAudience: List<String>,
-    val aivenBroker: String,
-    val truststoreLocation: String,
-    val keystoreLocation: String,
-    val credstorePassword: String,
-    var commonEnv: CommonEnvironment
+    val loginserviceAudience: List<String>
 )
 
-data class JobEnvironment(
-    val toggles: Toggles,
-    var commonEnv: CommonEnvironment,
+data class UrlEnv(
+    val syfosyketilfelleUrl: String,
+    val syfosmregisterUrl: String,
+    val syfosmregisterScope: String,
     val baseUrlDittSykefravaer: String,
-    val prometheusPushGatewayUrl: String
+    val stsUrl: String,
+    val pdlUrl: String,
+    val dkifUrl: String
 )
 
-data class CommonEnvironment(
-    val remote: Boolean,
+data class KafkaEnv(
     var kafkaBootstrapServersUrl: String,
     val kafkaSchemaRegistryUrl: String,
-    val stsUrl: String,
-    val pdlUrl: String,
-    val dkifUrl: String,
-    val serviceuserUsername: String,
-    val serviceuserPassword: String,
-    val dbEnvironment: DbEnvironment
+    val aivenBroker: String,
+    val sslConfig: KafkaSslEnv
 )
 
-data class DbEnvironment(
-    val databaseUrl: String,
-    val databaseName: String,
-    val dbVaultMountPath: String
+data class KafkaSslEnv(
+    val truststoreLocation: String,
+    val keystoreLocation: String,
+    val credstorePassword: String
 )
 
-data class Toggles(
-    val startJobb: Boolean,
+data class DbEnv(
+    var dbHost: String,
+    var dbPort: String,
+    var dbName: String,
+    val dbUsername: String = "",
+    val dbPassword: String = "",
+    val dbCredMounthPath: String = ""
+)
+
+data class ToggleEnv(
     val sendMerVeiledningVarsler: Boolean,
     val sendAktivitetskravVarsler: Boolean
 )
 
+data class JobEnv(
+    val sendVarsler: Boolean,
+    val jobTriggerUrl: String,
+    val serviceuserUsername: String,
+    val serviceuserPassword: String
+)
+
 fun getEnvVar(varName: String, defaultValue: String? = null) =
     System.getenv(varName) ?: defaultValue ?: throw RuntimeException("Missing required variable \"$varName\"")
 
+fun isGCP(): Boolean = getEnvVar("NAIS_CLUSTER_NAME").contains("gcp")
+
 fun isLocal(): Boolean = getEnvVar("KTOR_ENV", "local") == "local"
 
-fun isJob(): Boolean = getEnvVar("SEND_VARSLER", "NEI") == "JA"
+fun isJob(): Boolean = getEnvVar("JOB", "NEI") == "JA"
 
 private fun String.tilBoolean(): Boolean {
```

Comment:
Kan eventuelt bruke booleans direkte, og så lage en funksjon for å hente booleans. F.eks slik som dette, eller eventuelt med defaultValue dersom vi trenger det.

`fun getBooleanEnvVar(varName: String): () -> Boolean = {
    val variable = System.getenv(varName)

    if (variable === null) {
        throw RuntimeException("Missing required variable \"$varName\"")
    }

    variable.toBoolean()
}`

Reply:
Fikset